src/main/java/com/rog/EShop/services/UserService.java

```diff
@@ -38,35 +45,93 @@ public UserDto findById(Long id) {
         return userMapper.toDTO(user);
     }
 
+
     public UserDto save(UserRegisterDto userRegisterDto) {
 
         User user = userMapper.toEntity(userRegisterDto);
         if (userRepository.existsUserByUsername(user.getUsername())) {
             throw new ConflictException("This username already exists!");
         }
-
         if (!Objects.equals(userRegisterDto.getPassword(), userRegisterDto.getConfirmPassword())) {
             throw new BadRequestException("Password does not match!");
         }
-        String encode = bCryptPasswordEncoder.encode(userRegisterDto.getPassword());
-        user.setPassword(encode);
-        user.setRegisterDate(LocalDateTime.now());
-        user.setAccountNonExpired(true);
-        user.setAccountNonLocked(true);
-        user.setCredentialsNonExpired(true);
-        user.setEnabled(true);
-        user.setAuthorities(List.of(Role.ROLE_USER));
-        User userSaved = userRepository.save(user);
-        return userMapper.toDTO(userSaved);
 
+        ResponseTokenDto responseTokenDto = getTokenFromKeycloak();
+        ResponseEntity<String> response = getResponseFromKeycloak(userRegisterDto, responseTokenDto);
+        if (response.getStatusCode().isError()) {
+            throw new RuntimeException("Error " + response.getStatusCode());
+        }
+
+        ResponseEntity<List<ResponseKeycloakDto>> responseKeycloakDto = getResponseKeycloakDto(userRegisterDto,
+                responseTokenDto);
+
+        List<ResponseKeycloakDto> responseKeycloakDto1 = responseKeycloakDto.getBody();
+        user.setKeycloakId(responseKeycloakDto1.get(0).getId());
+        Long timeStamp = responseKeycloakDto1.get(0).getCreatedTimestamp();
+        user.setRegisterDate(LocalDateTime.ofInstant(Instant.ofEpochMilli(timeStamp), ZoneOffset.UTC));
+        userRepository.save(user);
+        return userMapper.toDTO(user);
 
     }
 
-    @Override
-    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
-        return userRepository.findByUsername(username)
-                .orElseThrow(() -> new UsernameNotFoundException("The user " + username + " does not exist"));
+    private static ResponseEntity<List<ResponseKeycloakDto>> getResponseKeycloakDto(UserRegisterDto userRegisterDto,
```

Comment:
Insead of input parameter `UserRegisterDto userRegisterDto` - Use `String username`
Insead of input parameter `ResponseTokenDto responseTokenDto` - Use `String accessToken`
Function name `getResponseKeycloakDto()` is also not understandable. Use `getUserFromKeycloakByUsername(String username, String accessToken)`.